examples/fluids/src/setupstg_shur14.c

```diff
@@ -0,0 +1,331 @@
+//
+// This file is part of CEED:  http://github.com/ceed
+
+/// @file
+/// Implementation of the Synthetic Turbulence Generation (STG) algorithm
+/// presented in Shur et al. 2014
+
+#include <stdlib.h>
+#include <math.h>
+#include <petsc.h>
+#include "../navierstokes.h"
+#include "setupstg_shur14.h"
+
+#ifndef M_PI
+#define M_PI    3.14159265358979323846
+#endif
+
+/*
+ * @brief Perform Cholesky decomposition on array of symmetric 3x3 matrices
+ *
+ * This assumes the input matrices are in order [11,22,33,12,13,23]. This
+ * format is also used for the output.
+ *
+ * @param[in]  comm   MPI_Comm
+ * @param[in]  nprofs Number of matrices in Rij
+ * @param[in]  Rij    Array of the symmetric matrices [6,nprofs]
+ * @param[out] Cij    Array of the Cholesky Decomposition matrices, [6,nprofs]
+ */
+PetscErrorCode CalcCholeskyDecomp(MPI_Comm comm, int nprofs,
+                                  const CeedScalar Rij[6][nprofs], CeedScalar Cij[6][nprofs]) {
+
+  PetscFunctionBeginUser;
+  for(int i=0; i<nprofs; i++) {
+    Cij[0][i] = sqrt(Rij[0][i]);
+    Cij[3][i] = Rij[3][i] / Cij[0][i];
+    Cij[1][i] = sqrt(Rij[1][i] - pow(Cij[3][i], 2) );
+    Cij[4][i] = Rij[4][i] / Cij[0][i];
+    Cij[5][i] = (Rij[5][i] - Cij[3][i]*Cij[4][i]) / Cij[1][i];
+    Cij[2][i] = sqrt(Rij[2][i] - pow(Cij[4][i], 2) - pow(Cij[5][i], 2));
+
+    if (isnan(Cij[0][i]) || isnan(Cij[1][i]) || isnan(Cij[2][i]))
+      SETERRQ(comm, -1, "Cholesky decomposition failed at profile point %d. "
+              "Either STGInflow has non-SPD matrix or contains nan.", i+1);
+  }
+  PetscFunctionReturn(0);
+}
+
+
+/*
+ * @brief Open a PHASTA *.dat file, grabbing dimensions and file pointer
+ *
+ * This function opens the file specified by `path` using `PetscFOpen` and
+ * passes the file pointer in `fp`. It is not closed in this function, thus
+ * `fp` must be closed sometime after this function has been called (using
+ * `PetscFClose` for example).
+ *
+ * Assumes that the first line of the file has the number of rows and columns
+ * as the only two entries, separated by a single space
+ *
+ * @param[in] comm MPI_Comm for the program
+ * @param[in] path Path to the file
+ * @param[in] char_array_len Length of the character array that should contain each line
+ * @param[out] dims Dimensions of the file, taken from the first line of the file
+ * @param[out] fp File pointer to the opened file
+ */
+static PetscErrorCode OpenPHASTADatFile(const MPI_Comm comm,
+                                        const char path[PETSC_MAX_PATH_LEN], const int char_array_len, int dims[2],
+                                        FILE **fp) {
+  PetscErrorCode ierr;
+  int ndims;
+  char line[char_array_len];
+  char **array;
+
+  PetscFunctionBeginUser;
+  ierr = PetscFOpen(comm, path, "r", fp); CHKERRQ(ierr);
+  ierr = PetscSynchronizedFGets(comm, *fp, char_array_len, line); CHKERRQ(ierr);
+  ierr = PetscStrToArray(line, ' ', &ndims, &array); CHKERRQ(ierr);
+  if(ndims != 2) SETERRQ(comm, -1,
```

Comment:
The spacing here  looks funny. (need space for the `if (` and the stuff on the following line doesn't match up) I'd make sure 'make style' is touching these files